test/JDBC/expected/cast_eliminate-vu-verify.out

```diff
@@ -120,3 +120,343 @@ text
+GO
+~~START~~
+int#!#smallint#!#bigint
+2147483647#!#2393#!#1111111111111111
+~~END~~
+
+~~START~~
+text
+Query Text: SELECT * FROM cast_eliminate WHERE (CAST(CAST(ROID AS BIGINT) AS BIGINT) = 3 OR CAST(CAST(ROID AS INT) AS BIGINT) = 2147483647) AND CAST(CAST(s_int AS INT) AS BIGINT) != 32767
+Bitmap Heap Scan on cast_eliminate (actual rows=1 loops=1)
+  Recheck Cond: ((roid = 3) OR (roid = 2147483647))
+  Filter: (((s_int)::integer)::bigint <> 32767)
+  Rows Removed by Filter: 1
+  Heap Blocks: exact=1
+  ->  BitmapOr (actual rows=0 loops=1)
+        ->  Bitmap Index Scan on cast_eliminate_pkey (actual rows=1 loops=1)
+              Index Cond: (roid = 3)
+        ->  Bitmap Index Scan on cast_eliminate_pkey (actual rows=1 loops=1)
+              Index Cond: (roid = 2147483647)
+~~END~~
+
+
+SELECT * FROM cast_eliminate WHERE CAST(CAST(s_int AS INT) AS BIGINT) >= 2394 AND CAST(CAST(ROID AS BIGINT) AS SMALLINT) >= 3
+GO
+~~START~~
+int#!#smallint#!#bigint
+3#!#32767#!#9223372036854775807
+~~END~~
+
+~~START~~
+text
+Query Text: SELECT * FROM cast_eliminate WHERE CAST(CAST(s_int AS INT) AS BIGINT) >= 2394 AND CAST(CAST(ROID AS BIGINT) AS SMALLINT) >= 3
+Seq Scan on cast_eliminate (actual rows=1 loops=1)
+  Filter: ((s_int >= 2394) AND (((roid)::bigint)::smallint >= 3))
+  Rows Removed by Filter: 4
+~~END~~
+
+
+SELECT * FROM cast_eliminate WHERE -32768 = CAST(s_int AS INT);
+GO
+~~START~~
+int#!#smallint#!#bigint
+-2147483648#!#-32768#!#-9223372036854775808
+~~END~~
+
+~~START~~
+text
+Query Text: SELECT * FROM cast_eliminate WHERE -32768 = CAST(s_int AS INT)
+Seq Scan on cast_eliminate (actual rows=1 loops=1)
+  Filter: ('-32768'::integer = s_int)
+  Rows Removed by Filter: 4
+~~END~~
+
+
+SELECT * FROM cast_eliminate WHERE -32768 = CAST(CAST(s_int AS INT) AS BIGINT);
+GO
+~~START~~
+int#!#smallint#!#bigint
+-2147483648#!#-32768#!#-9223372036854775808
+~~END~~
+
+~~START~~
+text
+Query Text: SELECT * FROM cast_eliminate WHERE -32768 = CAST(CAST(s_int AS INT) AS BIGINT)
+Seq Scan on cast_eliminate (actual rows=1 loops=1)
+  Filter: ('-32768'::integer = s_int)
+  Rows Removed by Filter: 4
+~~END~~
+
+
+SELECT * FROM cast_eliminate WHERE -32768 < CAST(CAST(s_int AS INT) AS BIGINT) OR CAST(ROID AS BIGINT) = -2147483648
+GO
+~~START~~
+int#!#smallint#!#bigint
+1#!#1#!#1
+2#!#2938#!#2
+3#!#32767#!#9223372036854775807
+-2147483648#!#-32768#!#-9223372036854775808
+2147483647#!#2393#!#1111111111111111
+~~END~~
+
+~~START~~
+text
+Query Text: SELECT * FROM cast_eliminate WHERE -32768 < CAST(CAST(s_int AS INT) AS BIGINT) OR CAST(ROID AS BIGINT) = -2147483648
+Seq Scan on cast_eliminate (actual rows=5 loops=1)
+  Filter: (('-32768'::integer < s_int) OR (roid = '-2147483648'::integer))
+~~END~~
+
+
+-- Throws an error due to overflow. If we wrongly eliminate the CAST function, the error won't be thrown
+SELECT * FROM cast_eliminate WHERE -32768 < CAST(CAST(s_int AS INT) AS BIGINT) OR -9223372036854775808 = CAST(CAST(b_int AS BIGINT) AS INT)
```

Comment:
Could you add the below test case to make sure removing a cast does not make an error?
```
SELECT * FROM cast_eliminate WHERE -9223372036854775808 = CAST(s_int AS BIGINT)
```

Reply:
Sure, added.